hedera-node/src/main/java/com/hedera/services/utils/TxProcessorUtil.java

```diff
@@ -0,0 +1,208 @@
+		return senderAccount.getMutable();
+	}
+
+	public static MutableAccount getMutableRelayer(HederaWorldState.Updater updater, Account relayer) {
+		final var relayerAccount = updater.getOrCreateSenderAccount(relayer.getId().asEvmAddress());
+		return relayerAccount.getMutable();
+	}
+
+	public static long calculateRefund(long gasLimit, long gasPrice, long gasUsedByTransaction,
+									   long sbhRefund, MutableAccount sender) {
+		final long refunded = gasLimit - gasUsedByTransaction + sbhRefund;
+		final Wei refundedWei = Wei.of(refunded * gasPrice);
+
+		if (refundedWei.greaterThan(Wei.ZERO)) {
+			sender.incrementBalance(refundedWei);
+		}
+		return refunded;
+	}
+
+	public static long calculateRefundEth(long gasLimit, long gasPrice, MutableAccount
+			mutableRelayer, Wei allowanceCharged, long sbhRefund, MutableAccount sender,
+										  long gasUsedByTransaction) {
+		final long refunded = gasLimit - gasUsedByTransaction + sbhRefund;
+		final Wei refundedWei = Wei.of(refunded * gasPrice);
+
+		if (refundedWei.greaterThan(Wei.ZERO)) {
+			if (allowanceCharged.greaterThan(Wei.ZERO)) {
+				// If allowance has been charged, we always try to refund relayer first
+				if (refundedWei.greaterOrEqualThan(allowanceCharged)) {
+					mutableRelayer.incrementBalance(allowanceCharged);
+					sender.incrementBalance(refundedWei.subtract(allowanceCharged));
+				} else {
+					mutableRelayer.incrementBalance(refundedWei);
+				}
+			} else {
+				sender.incrementBalance(refundedWei);
+			}
+		}
+		return refunded;
+	}
+
+	public static void sendFeesToCoinbase(HederaWorldState.Updater updater, GlobalDynamicProperties dynamicProperties,
+										  long gasLimit, long gasPrice, long refunded) {
+		final long coinbaseFee = gasLimit - refunded;
+		MutableAccount mutableCoinbase = updater.getOrCreate(
+				Id.fromGrpcAccount(dynamicProperties.fundingAccount()).asEvmAddress()).getMutable();
+		mutableCoinbase.incrementBalance(Wei.of(coinbaseFee * gasPrice));
+	}
+
+
+	public static TransactionProcessingResult buildTransactionResult(boolean isCompletedSuccess, List<Log> logs,
+																	 long sbhRefund, long gasUsedByTransaction,
+																	 long gasPrice, Address mirrorReceiver,
+																	 Bytes output, Optional<Bytes> revertReason,
+																	 Optional<ExceptionalHaltReason> exceptionalHaltReason,
+																	 Map<Address, Map<Bytes, Pair<Bytes, Bytes>>> stateChanges) {
+		if (isCompletedSuccess) {
+			return TransactionProcessingResult.successful(
+					logs,
+					gasUsedByTransaction,
+					sbhRefund,
+					gasPrice,
+					output,
+					mirrorReceiver,
+					stateChanges);
+		} else {
+			return TransactionProcessingResult.failed(
+					gasUsedByTransaction,
+					sbhRefund,
+					gasPrice,
+					revertReason,
+					exceptionalHaltReason,
+					stateChanges);
+		}
+	}
+
+	private TxProcessorUtil() {
+		throw new IllegalStateException("Utility class");
```

Comment:
If you change this to throw an `UnsupportedOperationException` (e.g. [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/AbiConstants.java#L29)), and then add `TxProcessorUtil.class` to [this list](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/test/java/com/hedera/services/utils/UtilsConstructorTest.java#L100), you get automatic coverage of such constructors.